File: flutter/feature_dashboard/lib/src/module/dashboard_module.dart
import 'package:contracts/contracts.dart';

/// Dashboard feature module — accessible to ADMIN, STAFF, and CUSTOMER.
class DashboardModule extends AppModule {
  @override
  ModuleMetadata get metadata => const ModuleMetadata(
        id: 'dashboard',
        name: 'Dashboard',
        requiredRoles: {Role.ADMIN, Role.STAFF, Role.CUSTOMER},
        priority: 100, // loads first
      );

  @override
  void initialize(ModuleContext context) {
    // Register widgets into slots
    for (final slot in provideWidgets()) {
      context.slotRegistry.register(slot);
    }
  }

  @override
  List<ModuleRoute> provideRoutes() => [
        const ModuleRoute(route: '/dashboard', requiredRole: Role.GUEST),
      ];
}

File: ios/Packages/FeatureDashboard/Sources/FeatureDashboard/Module/DashboardModule.swift
import Contracts
import WireCore

/// Dashboard feature module — accessible to ADMIN, STAFF, and CUSTOMER.
public final class DashboardModule: AppModule {
    public init() {}

    public let metadata = ModuleMetadata(
        id: "dashboard",
        name: "Dashboard",
        requiredRoles: [.ADMIN, .STAFF, .CUSTOMER],
        priority: 100 // loads first
    )

    public func initialize(context: any ModuleContext) {
        for slot in provideWidgets() {
            context.slotRegistry.register(slot)
        }
    }

    public func provideRoutes() -> [ModuleRoute] {
        [ModuleRoute(route: "/dashboard", requiredRole: .GUEST)]
    }
}

